File: backEnd_flask/models/movie_model.py
from pydantic import BaseModel
from typing import List, Optional

class Movie(BaseModel):
    id: Optional[str]
    title: Optional[str]
    year: Optional[str]
    certificate: Optional[str]
    runtime: Optional[str]
    genres: Optional[List[str]]
    imdb_rating: Optional[str]
    meta_score: Optional[str]
    director: Optional[str]
    cast: Optional[List[str]]
    votes: Optional[str]
    gross: Optional[str]

File: NetflixClone/src/app/components/model/movie.model.ts
export interface Movie {
  id: string;
  title: string;
  year: number;
  certificate?: string;
  runtime: number;
  genres: string[];
  imdb_rating: number;
  meta_score?: number;
  director: string;
  cast: string[];
  votes: number;
  poster: string;
  gross?: number;
  image_url: string;
  users_review: number;
}
